torch/_inductor/codegen/common.py

```diff
@@ -1770,9 +1770,12 @@ def bucketize(
         self,
         values: CSEVariable,
         offsets_name: str,
+        num_bucket_boundaries: sympy.Expr,
         offsets_size: sympy.Expr,
         indexing_dtype: torch.dtype,
         right: bool,
+        offsets_indices: CSEVariable,
+        sorter_name: Optional[str] = None,
```

Comment:
is it possible to make these changes in a backward-compatible way?

problem 1: I think `torch/_inductor/jagged_lowerings.py` needs to be updated
problem 2: there are some internal lowerings that use `ops.bucketize` that would need to be updated simultaneously.

Note: if there's a reason why we can't make this backward-compatible, it's not a deal breaker (for (1) you'll just need to update the lowerings, unless I'm missing something; and for (2) I can help make the internal changes simultaneously). But if possible, can we just put `num_bucket_boundaries` at the bottom with the other new paramett and give it a default value?

cc @jbschlosser do you know if there are tests that should have caught this or did target determinator just skip them?

Reply:
@davidberard98 I can look into that; I was operating under the assumption that I could see any lowerings that used this, since it was an internal API (my bad!). The main difficulty that I think we'll have with backwards compatibility is the assumption of a single dimension; having a sorter array present is trivial to do backwards compatibly.

To fix the striding issue, I was thinking about tinkering with this again, so let me give it more consideration. Maybe I can come up with a set of changes that'll be minimally disruptive, even if not fully BC.

Reply:
Thanks for the pointer to `jagged_lowerings`, btw; I had indeed missed that one.

Reply:
if you just put the new parameters at the end of the function signature and give them default arguments that match the previous behavior, would that work?

Reply:
For this MR, sure. Would you be open to changing the function signature in the near future, though? There'll be at least two more parameters added for the striding, and this signature is already pretty complicated for my tastes.

Reply:
After some offline discussion with @davidberard98, we've agreed that due to the striding bug, the API for `ops.bucketize` will have to change in non-backwards-compatible ways in order to force all consumers of the API to handle striding. To minimize the impact on Meta's internal work, I'll make that fix in this PR, such that only one set of changes will need to be done.